Implement FrozenSet constructors, __len__

ASP1234 opened the pull request:
**Changes**

- Implemented FrozenSet Constructor as per the requirement

- Added __len__ method to finish the builtin len() fn

**Testing**

- Pass all tests for len()

- Also added sample data to frozenset test in utils.py . New implementation passes the test data successfully

cflee commented on tests/utils.py:
```diff
@@ -647,6 +647,9 @@ def wrapper(*args, **kwargs):
             'frozenset({1, 2.3456, "another"})',
+            'frozenset([1, 2.3456, "another"])',
+            'frozenset((1, 2.3456, "another"))',
+            'frozenset(iter([1, 2.3456, "another"]))',
```
It looks like you're trying to test the frozenset constructor here. This is not the best place to do it, as the `SAMPLE_DATA` is meant to provide a variety of values, not a bunch of the same values.

Instead, these tests should go into `tests/datatypes/test_frozenset.py`, under the 'FrozensetTests' test case.

ASP1234 commented:
**Addressing PR comments**

- Fixed codestyle errors

- Moved tests to FrozenSetTests

- Added behavior for non-iterable i/p

cflee commented on tests/datatypes/test_frozenset.py:
```diff
@@ -2,7 +2,33 @@
+        self.assertCodeExecution("""
+            x = frozenset('a')
+            print(x)
+            """)
```
Please combine all these tests into one large `self.assertCodeExecution()`, it's ok to keep assigning the value of x and then accessing it.

cflee commented on tests/datatypes/test_frozenset.py:
```diff
@@ -2,7 +2,33 @@
+            """)
+
+        self.assertCodeExecution("""
+            x = frozenset((1, 2))
```
This has tested list and tuple, should there be a test for a set (and frozenset) as input too? We should also test whether changing the source set/array/tuple affects the newly created frozenset.

cflee commented on python/common/org/python/types/FrozenSet.java:
```diff
@@ -1,13 +1,82 @@
+    @org.python.Method(
+            __doc__ = ""
+    )
+    public org.python.types.Str __repr__() {
```
Since this method has been implemented, the `tests.builtins.test_repr.BuiltinReprFunctionTests.test_frozenset` test should be passing as well, could you check why it's still failing?

ASP1234 replied:
It's not failing. Just needed to remove from not implemented list :)

cflee commented on python/common/org/python/types/FrozenSet.java:
```diff
@@ -1,13 +1,82 @@
+                buffer.append(args[0].typeName());
+                buffer.append("' object is not iterable");
+
+                throw new org.python.exceptions.TypeError(buffer.toString());
```
Is this `else` case necessary, since `org.Python.iter()` will throw TypeError with this exact message if the parameter is not iterable?

ASP1234 replied:
Didn't know about this earlier. Removed the redundant else part

ASP1234 commented on python/common/org/python/types/FrozenSet.java:
```diff
@@ -1,13 +1,82 @@
-    )
+                    "Build an immutable unordered collection of unique elements.\n",
+            default_args = {"iterable"}
+            )
```
Fixed!

ASP1234 commented:
**Changes**

- Moved creation tests into a single tests

- Tested by changing source of frozenset, expected o/p should not change with source changes

- Implemented __repr__, thus tests pass for frozenset __repr__ 😄 

- Removed else part from constructors for non iterable objects. TypeError takes care of it it seems!

ASP1234 commented:
Checkstyle fix
